### PR TITLE
feat: Upgrade Typescript Dictionary

### PR DESCRIPTION
**Pull request by KieranRobson**
# Add/Fix Dictionary

Dictionary: Typescript

## Description

Upgrade as part of #705 

## References

NA

## Checklist

- [x] By submitting this issue, you agree to follow our
      [Code of Conduct](https://github.com/streetsidesoftware/cspell-dicts/blob/main/CODE_OF_CONDUCT.md)

----

When running `yarn run test`,  output is 

```
$ yarn run test
yarn run v1.22.19
$ head -n 100 src/typescript.txt | cspell -v -c ./cspell-ext.json "--local=*" " --languageId=typescript" stdin

cspell;
Date: Wed, 21 Dec 2022 08:29:58 GMT
Options:
    verbose:   Yes
    config:    ./cspell-ext.json
    exclude:   node_modules/**
    files:     --languageId=typescript,stdin
    wordsOnly: No
    unique:    No

Config Files Found:
    ./cspell-ext.json

Exclusion Globs: 
    Glob: node_modules/** from command line

1/1 .\stdinChecking: stdin, File type: auto, Language: *
Dictionary Error with (aws) Error: ENOENT: no such file or directory, open 'D:\Projects\Misc Project\cspell-dicts\dictionaries\aws\aws.txt.gz'
Dictionary Error with (cryptocurrencies) Error: ENOENT: no such file or directory, open 'D:\Projects\Misc Project\cspell-dicts\dictionaries\cryptocurrencies\cryptocurrencies.txt.gz'
Dictionary Error with (filetypes) Error: ENOENT: no such file or directory, open 'D:\Projects\Misc Project\cspell-dicts\dictionaries\filetypes\filetypes.txt.gz'
Dictionary Error with (public-licenses) Error: ENOENT: no such file or directory, open 'D:\Projects\Misc Project\cspell-dicts\dictionaries\public-licenses\public-licenses.txt.gz'
Checked: stdin, File type: text, Language: * ... Issues: 90 0.223546S
Config file Used: D:\Projects\Misc Project\cspell-dicts\dictionaries\typescript\cspell-ext.json
Dictionaries Used: companies, filetypes, public-licenses, softwareTerms, aws, cryptocurrencies
 223.55ms X
\:1:1 - Unknown word (aaguid)
\:2:1 - Unknown word (abbr)
\:3:1 - Unknown word (abort)
\:4:1 - Unknown word (aborted)
\:6:1 - Unknown word (absent)
\:7:1 - Unknown word (absolute)
\:8:1 - Unknown word (abstract)
\:9:1 - Unknown word (acceleration)
\:10:1 - Unknown word (accelerometer)
\:11:1 - Unknown word (accept)
\:12:1 - Unknown word (access)
\:13:1 - Unknown word (accessibility)
\:15:1 - Unknown word (account)
\:16:1 - Unknown word (accumulate)
\:17:1 - Unknown word (accuracy)
\:19:1 - Unknown word (acosh)
\:20:1 - Unknown word (acquire)
\:21:1 - Unknown word (acquisition)
\:22:1 - Unknown word (across)
\:23:1 - Unknown word (action)
\:24:1 - Unknown word (actions)
\:25:1 - Unknown word (activatable)
\:26:1 - Unknown word (activate)
\:27:1 - Unknown word (activated)
\:28:1 - Unknown word (activating)
\:29:1 - Unknown word (active)
\:30:1 - Unknown word (activity)
\:31:1 - Unknown word (actual)
\:32:1 - Unknown word (actuator)
\:33:1 - Unknown word (actuators)
\:35:1 - Unknown word (added)
\:36:1 - Unknown word (addition)
\:37:1 - Unknown word (additional)
\:38:1 - Unknown word (additive)
\:40:1 - Unknown word (address)
\:41:1 - Unknown word (addsourcebuffer)
\:42:1 - Unknown word (addtrack)
\:43:1 - Unknown word (adjacent)
\:44:1 - Unknown word (adjust)
\:45:1 - Unknown word (adopt)
\:48:1 - Unknown word (advise)
\:49:1 - Unknown word (affect)
\:50:1 - Unknown word (affected)
\:51:1 - Unknown word (after)
\:52:1 - Unknown word (afterbegin)
\:53:1 - Unknown word (afterend)
\:54:1 - Unknown word (afterprint)
\:55:1 - Unknown word (agent)
\:56:1 - Unknown word (aggregate)
\:57:1 - Unknown word (aggregated)
\:58:1 - Unknown word (aggressive)
\:59:1 - Unknown word (ahead)
\:60:1 - Unknown word (alert)
\:61:1 - Unknown word (algorithm)
\:62:1 - Unknown word (algorithms)
\:63:1 - Unknown word (alias)
\:64:1 - Unknown word (aliased)
\:65:1 - Unknown word (aliasing)
\:67:1 - Unknown word (alignment)
\:68:1 - Unknown word (alink)
\:70:1 - Unknown word (allocate)
\:71:1 - Unknown word (allow)
\:72:1 - Unknown word (allowed)
\:73:1 - Unknown word (allpass)
\:74:1 - Unknown word (alpha)
\:75:1 - Unknown word (alphabetic)
\:76:1 - Unknown word (already)
\:77:1 - Unknown word (alternate)
\:78:1 - Unknown word (alternative)
\:79:1 - Unknown word (alternatives)
\:80:1 - Unknown word (altitude)
\:81:1 - Unknown word (always)
\:82:1 - Unknown word (ambient)
\:83:1 - Unknown word (amount)
\:84:1 - Unknown word (ampersand)
\:85:1 - Unknown word (amplitude)
\:86:1 - Unknown word (analyser)
\:87:1 - Unknown word (ancestor)
\:88:1 - Unknown word (anchor)
\:89:1 - Unknown word (anchors)
\:91:1 - Unknown word (angle)
\:92:1 - Unknown word (angletype)
\:93:1 - Unknown word (angular)
\:94:1 - Unknown word (anim)
\:95:1 - Unknown word (animatable)
\:96:1 - Unknown word (animate)
\:97:1 - Unknown word (animated)
\:98:1 - Unknown word (animation)
\:99:1 - Unknown word (animationcancel)
\:100:1 - Unknown word (animationend)
CSpell: Files checked: 1, Issues found: 90 in 1 files
error Command failed with exit code 1.
````

This is because of 
````
Dictionary Error with (aws) Error: ENOENT: no such file or directory, open 'D:\Projects\Misc Project\cspell-dicts\dictionaries\aws\aws.txt.gz'
Dictionary Error with (cryptocurrencies) Error: ENOENT: no such file or directory, open 'D:\Projects\Misc Project\cspell-dicts\dictionaries\cryptocurrencies\cryptocurrencies.txt.gz'
Dictionary Error with (filetypes) Error: ENOENT: no such file or directory, open 'D:\Projects\Misc Project\cspell-dicts\dictionaries\filetypes\filetypes.txt.gz'
Dictionary Error with (public-licenses) Error: ENOENT: no such file or directory, open 'D:\Projects\Misc Project\cspell-dicts\dictionaries\public-licenses\public-licenses.txt.gz'
````

Thought id still do the PR incase this error goes when all dicts are built

### CODE REVIEW

**Comment by Jason3S** [comment]
@KieranRobson,

Thank you. Those files won't exist if `yarn` wasn't run before running `yarn test`.

```sh
yarn
yarn test
```

File: dictionaries/typescript/package.json [hunk 1]
```diff
@@ -11,8 +11,8 @@
     "./cspell-ext.json": "./cspell-ext.json"
   },
   "scripts": {
-    "build": "cspell-tools compile \"typescript.txt\" --no-compress -o ./dict",
-    "test": "head -n 100 \"typescript.txt\" | cspell -v -c ./cspell-ext.json \"--local=*\" \"--languageId=typescript\" stdin",
+    "build": "cspell-tools-cli compile --keep-raw-case src/typescript.txt --no-compress -o ./dict",
+    "test": "head -n 100 src/typescript.txt | cspell -v -c ./cspell-ext.json \"--local=*\" \" --languageId=typescript\" stdin",
```
Comment (Jason3S) [inline dictionaries/typescript/package.json:15]: `\" --languageId=typescript\"` should be `\"--languageId=typescript\"

**Comment by Jason3S** [comment]
@KieranRobson,

if you do:

from `dictionarys/typescript`
```sh
git mv typescript.txt src/typescript.txt
```

The diff will me much smaller.

**Comment by KieranRobson** [comment]
Thanks @Jason3S, that fixed the error!

Running `git mv typescript.txt src/typescript.txt` led to 
`fatal: destination exists, source=dictionaries/typescript/typescript.txt, destination=dictionaries/typescript/src/typescript.txt`

**Comment by Jason3S** [comment]
> Thanks @Jason3S, that fixed the error!
> 
> Running `git mv typescript.txt src/typescript.txt` led to `fatal: destination exists, source=dictionaries/typescript/typescript.txt, destination=dictionaries/typescript/src/typescript.txt`

It is complaining because you already added it.
You have to remove the "new" one first.

```sh
git rm src/typescript.txt
git mv typescript.txt src/typescript.txt
```

**Comment by Jason3S** [comment]
Thank you!

**Pull request merged**